src/input.rs

```diff
@@ -814,6 +818,23 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
 
         has_binding
     }
+
+    // TODO: Handle multi-line message bars
+    /// Handle clicks on the message bar.
+    ///
+    /// Returns `true` if the click was within the message bar.
+    fn on_message_bar_click(&mut self) -> bool {
```

Comment:
It's a tiny point, but it's why I was proposing three functions here. I find it odd to be calling a function named `on_message_bar_click` even when we're not clicking on a message bar. Maybe this function could be named something like `maybe_on_message_bar_click` or `when_on_message_bar_click`, but ehhhhh.

If you break the conditional logic out into a `mouse_over_message_bar` boolean function, and then have the `on_message_bar_click` take the `point` and just do the following bit, I think it would be better.

```rs
if point.col + message_bar::CLOSE_BUTTON_TEXT.len() >= self.ctx.size_info().cols() {
    self.ctx.message_bar().pop();
}
```